src/main/java/duke/TaskManager.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import duke.task.Task;
+
+/**
+ * Represents a TaskManager object that stores and operates on a List of tasks.
+ */
+public class TaskManager {
+    private List<Task> taskList;
+    private final Storage storage = new Storage();
+
+    TaskManager() throws DateTimeException {
+        load();
+    }
+
+    /**
+     * Adds a task to the taskList and saves it to database.
+     *
+     * @param task any Task object is accepted.
+     */
+    public void addTask(Task task) {
+        taskList.add(task);
+        save();
+    }
+
+    /**
+     * Deletes a task at specified index.
+     * If given index is too big or negative, an IllegalArgumentException is thrown.
+     *
+     * @param taskId The index (0-indexed) of the task to be deleted.
+     * @return The task that has just been deleted.
+     * @throws IllegalArgumentException If taskId >= taskList.size() or taskId < 0.
+     */
+    public Task deleteTask(int taskId) throws IllegalArgumentException {
+        if (taskId < taskList.size() && taskId >= 0) {
+            Task ret = taskList.remove(taskId);
+            save();
+            return ret;
+        } else {
+            throw new IllegalArgumentException("☹ OOPS!!! Task Index is invalid!!");
+        }
+    }
+
+    public int taskCount() {
```

Comment:
```suggestion
    public int getNumOfTasks() {
```
Perhaps you could name methods in verb form? For e.g. `getNumOfTasks`, `countTasks`, `getSize`